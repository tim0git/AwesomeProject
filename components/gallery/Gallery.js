import React from 'react';
// import PropTypes from 'prop-types';
import {
  SafeAreaView,
  View,
  ScrollView,
  StyleSheet,
  Text,
  TouchableOpacity,
  Image,
} from 'react-native';

import images from '../../resources/images/index';

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    justifyContent: 'space-around',
    alignItems: 'center',
    height: '100%',
    textAlign: 'center',
  },
  description: {
    textAlign: 'center',
  },
});

function Gallery({navigation}) {
  // can i not map through the above? is this the flat list component? are there performance improvements using flat list?
  return (
    <SafeAreaView>
      <ScrollView contentContainerStyle={styles.container}>
        <View>
          <TouchableOpacity onPress={() => navigation.navigate('Forrest')}>
            <Image
              source={images.forrest}
              style={{
                resizeMode: 'cover',
                height: 150,
                width: 200,
              }}
            />
            <Text style={styles.description}>Forrest</Text>
          </TouchableOpacity>
        </View>
        <View>
          <TouchableOpacity onPress={() => navigation.navigate('Beach')}>
            <Image
              source={images.beach}
              style={{
                resizeMode: 'cover',
                height: 150,
                width: 200,
              }}
            />
            <Text style={styles.description}>Beach</Text>
          </TouchableOpacity>
        </View>
        <View>
          <TouchableOpacity onPress={() => navigation.navigate('City')}>
            <Image
              source={images.city}
              style={{
                resizeMode: 'cover',
                height: 150,
                width: 200,
              }}
            />
            <Text style={styles.description}>City</Text>
          </TouchableOpacity>
        </View>
        <View>
          <TouchableOpacity onPress={() => navigation.navigate('Mountains')}>
            <Image
              source={images.mountains}
              style={{
                resizeMode: 'cover',
                height: 150,
                width: 200,
              }}
            />
            <Text style={styles.description}>Mountains</Text>
          </TouchableOpacity>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}

Gallery.propTypes = {};

export default Gallery;
