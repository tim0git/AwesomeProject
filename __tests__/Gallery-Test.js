import 'react-native';
import React from 'react';
import 'jest-enzyme';
import 'react-native-mock-render/mock';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

// import App from '../App';
// import Hello from '../components/Hello';
// import Details from '../components/Details';
import Gallery from '../components/gallery/Gallery';

describe('<Gallery >', () => {
  const gallery = <Gallery />;

  it('Component Renders', () => {
    const component = shallow(gallery);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('displays 4 Images', () => {
    const count = [0, 1, 2, 3];
    const wrapper = shallow(gallery);
    count.forEach((number) => {
      expect(wrapper.find('Image').at(number)).toExist();
    });
  });

  it('Image has the id in the order sepcified in the array below.', () => {
    const wrapper = shallow(gallery);
    const imageIds = ['Forrest', 'Beach', 'City', 'Mountains'];
    imageIds.forEach((desc, index) => {
      expect(wrapper.find('Image').at(index).props().id).toBe(desc);
    });
  });

  it('View has the Text in the order sepcified in the array below.', () => {
    const wrapper = shallow(gallery);
    const imageIds = ['Forrest', 'Beach', 'City', 'Mountains'];
    imageIds.forEach((desc, index) => {
      expect(wrapper.find('Text').at(index).props().children).toBe(desc);
    });
  });

  it('Screen only displays 4 Images ', () => {
    const wrapper = shallow(gallery);
    expect(wrapper.find('Images').at(4).exists()).toBe(false);
  });

  it('Screen only displays 4 Text ', () => {
    const wrapper = shallow(gallery);
    expect(wrapper.find('Text').at(4).exists()).toBe(false);
  });

  it.todo('test the onPress funcitonality for each..');
});
