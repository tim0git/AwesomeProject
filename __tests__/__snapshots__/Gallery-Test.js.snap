// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Gallery > Component Renders 1`] = `
<ForwardRef(SafeAreaView)>
  <ScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "display": "flex",
        "height": "100%",
        "justifyContent": "space-around",
        "textAlign": "center",
      }
    }
  >
    <View
      key="Forrest"
    >
      <ForwardRef
        onPress={[Function]}
      >
        <Image
          id="Forrest"
          source="test-file-stub"
          style={
            Object {
              "height": 150,
              "resizeMode": "cover",
              "width": 200,
            }
          }
        />
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Forrest
        </Text>
      </ForwardRef>
    </View>
    <View
      key="Beach"
    >
      <ForwardRef
        onPress={[Function]}
      >
        <Image
          id="Beach"
          source="test-file-stub"
          style={
            Object {
              "height": 150,
              "resizeMode": "cover",
              "width": 200,
            }
          }
        />
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Beach
        </Text>
      </ForwardRef>
    </View>
    <View
      key="City"
    >
      <ForwardRef
        onPress={[Function]}
      >
        <Image
          id="City"
          source="test-file-stub"
          style={
            Object {
              "height": 150,
              "resizeMode": "cover",
              "width": 200,
            }
          }
        />
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          City
        </Text>
      </ForwardRef>
    </View>
    <View
      key="Mountains"
    >
      <ForwardRef
        onPress={[Function]}
      >
        <Image
          id="Mountains"
          source="test-file-stub"
          style={
            Object {
              "height": 150,
              "resizeMode": "cover",
              "width": 200,
            }
          }
        />
        <Text
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          Mountains
        </Text>
      </ForwardRef>
    </View>
  </ScrollView>
</ForwardRef(SafeAreaView)>
`;
